### DISCUSSION TITLE
Is there a guide to run top-down analysis on the vswarm benchmark?

### DISCUSSION THREAD

**Discussion by zhewenhu** [vhive-serverless/vSwarm#553] [Q&A]
I am currently referring to the [profiling guide](https://github.com/vhive-serverless/vHive/blob/main/docs/profiling.md), which provides support for a few functions. However, I am unsure how to proceed with profiling the vSwarm benchmark. First I followed the vHive quickstart guide to setup a single-node cluster with "stock-only" option, then I deploy the "hotel-app-geo" microbenchmark with knative, but the profiling tool does not recognize the geo function. Could you kindly provide guidance or direct me to any relevant documentation that outlines the specific procedures for profiling the vSwarm benchmark?

**Comment by ustiugov** [comment]
thank you for the question. I assume you would like to profile a container-only deployment (no VMs). You should profile the whole system using perf's `-a` option

@dhschall may be able to provide further details
